clang/lib/Basic/Targets/X86.cpp

```diff
@@ -1418,6 +1418,14 @@ bool X86TargetInfo::validateAsmConstraint(
   case 'O':
     Info.setRequiresImmediate(0, 127);
     return true;
+  case 'W':
+    switch (*++Name) {
+    default:
+      return false;
+    case 's':
+      Info.setAllowsRegister();
```

Comment:
Not sure if it matters to a print constraint, should it be `setAllowsMemory`?

Reply:
`setAllowsRegister` is somewhat confusing but its use is correct here (also used by aarch64/riscv `S`).

`setAllowsMemory` seems to not add extra checks.